patches/chrome-android-java-src-org-chromium-chrome-browser-bookmarks-BookmarkUndoController.java.patch

```diff
@@ -0,0 +1,43 @@
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkUndoController.java b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkUndoController.java
+index fad6de96cf7bd17a9eb7ab828cc97544cc94e422..a8384e71dc8935854b87ef4b6d85dac91a63656f 100644
+--- a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkUndoController.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkUndoController.java
+@@ -13,6 +13,7 @@ import org.chromium.chrome.browser.bookmarks.BookmarkModel.BookmarkDeleteObserve
+ import org.chromium.chrome.browser.snackbar.Snackbar;
+ import org.chromium.chrome.browser.snackbar.SnackbarManager;
+ 
++import java.util.List;
+ import java.util.Locale;
+ 
+ /**
+@@ -25,6 +26,7 @@ public class BookmarkUndoController extends BookmarkModelObserver implements
+     private final BookmarkModel mBookmarkModel;
+     private final SnackbarManager mSnackbarManager;
+     private final Context mContext;
++    private List<BookmarkItem> mBookmarks;
+ 
+     /**
+      * Creates an instance of {@link BookmarkUndoController}.
+@@ -56,6 +58,7 @@ public class BookmarkUndoController extends BookmarkModelObserver implements
+ 
+     @Override
+     public void onDismissNoAction(Object actionData) {
++        BraveBookmarkWorker.syncDeletedBookmarks(mBookmarks); if (mBookmarks != null) mBookmarks.clear();
+     }
+ 
+     // Overriding BookmarkModelObserver
+@@ -76,11 +79,12 @@ public class BookmarkUndoController extends BookmarkModelObserver implements
+ 
+     // Implement BookmarkDeleteObserver
+     @Override
+-    public void onDeleteBookmarks(String[] titles, boolean isUndoable) {
++    public void onDeleteBookmarks(String[] titles, List<BookmarkItem> bookmarks, boolean isUndoable) {
+         assert titles != null && titles.length >= 1;
+-
++        BraveBookmarkWorker.syncDeletedBookmarks(mBookmarks); if (mBookmarks != null) mBookmarks.clear();
```

Comment:
please add a new line instead of replacing the empty line

Reply:
done